Checkstyle: Javadoc not required on private method

Identified at https://github.com/checkstyle/checkstyle/pull/7932#discussion_r471834347 ,

All methods should have a javadoc in main code, including private methods. Somehow our config is broken that it is not reporting this.

I have identified the issue.

The scope between [JavadocMethodCheck](https://github.com/checkstyle/checkstyle/blob/43f500fda7ea85a358fcbd1585716f3effcc772d/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheck.java#L375) and [MissingJavadocMethodCheck](https://github.com/checkstyle/checkstyle/blob/8179a34d87881f480b1b26b44bafcbcc6607cec9/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingJavadocMethodCheck.java#L278) is different. Neither one is set in our config, https://github.com/checkstyle/checkstyle/blob/master/config/checkstyle_checks.xml .

This has been the default since we created [MissingJavadocMethodCheck](https://github.com/checkstyle/checkstyle/issues/6703#issue-438291993).

We just need to set the scope in MissingJavadocMethodCheck to private, the same as JavadocMethodCheck, and it will resolve the issue.

When fixed, the following violations come up.
````
[ERROR] [checkstyle] [ERROR] \src\it\java\org\checkstyle\base\AbstractItModuleTestSupport.java:315:5: Missing a Javadoc comment. [MissingJavadocMethod]
[ERROR] [checkstyle] [ERROR] \src\main\java\com\puppycrawl\tools\checkstyle\checks\coding\FinalLocalVariableCheck.java:389:5: Missing a Javadoc comment. [MissingJavadocMethod]
[ERROR] [checkstyle] [ERROR] \src\main\java\com\puppycrawl\tools\checkstyle\checks\coding\OneStatementPerLineCheck.java:279:5: Missing a Javadoc comment. [MissingJavadocMethod]
[ERROR] [checkstyle] [ERROR] \src\main\java\com\puppycrawl\tools\checkstyle\checks\coding\OneStatementPerLineCheck.java:291:5: Missing a Javadoc comment. [MissingJavadocMethod]
[ERROR] [checkstyle] [ERROR] \src\main\java\com\puppycrawl\tools\checkstyle\checks\coding\OneStatementPerLineCheck.java:297:5: Missing a Javadoc comment. [MissingJavadocMethod]
[ERROR] [checkstyle] [ERROR] \src\main\java\com\puppycrawl\tools\checkstyle\checks\indentation\LambdaHandler.java:121:5: Missing a Javadoc comment. [MissingJavadocMethod]
[ERROR] [checkstyle] [ERROR] \src\main\java\com\puppycrawl\tools\checkstyle\meta\MetadataGeneratorUtil.java:43:5: Missing a Javadoc comment. [MissingJavadocMethod]
[ERROR] [checkstyle] [ERROR] \src\main\java\com\puppycrawl\tools\checkstyle\meta\ModuleType.java:37:5: Missing a Javadoc comment. [MissingJavadocMethod]
````

We should review the scopes of the other Missing checks to ensure they match their other.

--- a/config/checkstyle_checks.xml
+++ b/config/checkstyle_checks.xml
@@ -570,9 +570,12 @@
     <module name="JavadocVariable"/>
     <module name="MissingJavadocMethod">
       <property name="allowMissingPropertyJavadoc" value="true"/>
+      <property name="scope" value="private"/>
     </module>
     <module name="MissingJavadocPackage"/>
-    <module name="MissingJavadocType"/>
+    <module name="MissingJavadocType">
+      <property name="scope" value="private"/>
+    </module>
     <module name="NonEmptyAtclauseDescription"/>
     <module name="SingleLineJavadoc"/>
     <module name="WriteTag"/>

--- a/src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java
+++ b/src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java
@@ -312,6 +312,12 @@ public abstract class AbstractItModuleTestSupport extends AbstractPathTestSuppor
         return checkMessage;
     }
 
+    /**
+     * Remove device from path string for windows path.
+     *
+     * @param path path to correct.
+     * @return Path without device name.
+     */
     private static String removeDeviceFromPathOnWindows(String path) {
         String fixedPath = path;
         final String os = System.getProperty("os.name", "Unix");

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MatchXpathCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MatchXpathCheck.java
@@ -270,6 +270,13 @@ public class MatchXpathCheck extends AbstractCheck {
         }
     }
 
+    /**
+     * Find nodes that match query.
+     *
+     * @param rootAST root node
+     * @return list of matching nodes
+     * @throws IllegalStateException if evaluation of xpath query fails
+     */
     private List<DetailAST> findMatchingNodesByXpathQuery(DetailAST rootAST) {
         try {
             final RootNode rootNode = new RootNode(rootAST);

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/OneStatementPerLineCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/OneStatementPerLineCheck.java
@@ -276,6 +276,12 @@ public final class OneStatementPerLineCheck extends AbstractCheck {
         }
     }
 
+    /**
+     * Checks semicolon placement in lambda.
+     *
+     * @param ast semicolon to check
+     * @param currentStatement current statement
+     */
     private void checkLambda(DetailAST ast, DetailAST currentStatement) {
         int countOfSemiInCurrentLambda = countOfSemiInLambda.pop();
         countOfSemiInCurrentLambda++;
@@ -288,12 +294,23 @@ public final class OneStatementPerLineCheck extends AbstractCheck {
         }
     }
 
+    /**
+     * Checks that given node is a resource.
+     *
+     * @param ast semicolon to check
+     * @return true if node is a resource
+     */
     private static boolean isResource(DetailAST ast) {
         return ast != null
             && (ast.getType() == TokenTypes.RESOURCES
                  || ast.getType() == TokenTypes.RESOURCE_SPECIFICATION);
     }
 
+    /**
+     * Checks resource variable.
+     *
+     * @param currentStatement current statement
+     */
     private void checkResourceVariable(DetailAST currentStatement) {
         if (treatTryResourcesAsStatement) {
             final DetailAST nextNode = currentStatement.getNextSibling();

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/LambdaHandler.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/LambdaHandler.java
@@ -115,6 +115,13 @@ public class LambdaHandler extends AbstractExpressionHandler {
         }
     }
 
+    /**
+     * Checks that given indent is acceptable or not.
+     *
+     * @param astColumnNo indent value to check
+     * @param level indent level
+     * @return true if indent is not acceptable
+     */
     private boolean isNonAcceptableIndent(int astColumnNo, IndentLevel level) {
         return astColumnNo < level.getFirstIndentLevel()
             || getIndentCheck().isForceStrictCondition()

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/LineWrappingHandler.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/LineWrappingHandler.java
@@ -305,6 +305,13 @@ public class LineWrappingHandler {
         return endOfScope;
     }
 
+    /**
+     * Checks that some parent of given node contains given token type.
+     *
+     * @param node node to check
+     * @param type type to look for
+     * @return true if there is a parent of given type
+     */
     private static boolean isParentContainsTokenType(final DetailAST node, int type) {
         boolean returnValue = false;
         for (DetailAST ast = node.getParent(); ast != null; ast = ast.getParent()) {

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/sizes/LambdaBodyLengthCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/sizes/LambdaBodyLengthCheck.java
@@ -187,6 +187,12 @@ public class LambdaBodyLengthCheck extends AbstractCheck {
         }
     }
 
+    /**
+     * Get length of lambda body.
+     *
+     * @param ast lambda body node.
+     * @return length of lambda body.
+     */
     private static int getLength(DetailAST ast) {
         final DetailAST lambdaBody = ast.getLastChild();
         final int length;

--- a/src/main/java/com/puppycrawl/tools/checkstyle/meta/MetadataGeneratorUtil.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/meta/MetadataGeneratorUtil.java
@@ -40,6 +40,7 @@ import com.puppycrawl.tools.checkstyle.utils.TokenUtil;
 /** Class which handles all the metadata generation and writing calls. */
 public final class MetadataGeneratorUtil {
 
+    /** Stop instances being created. **/
     private MetadataGeneratorUtil() {
     }
 

--- a/src/main/java/com/puppycrawl/tools/checkstyle/meta/ModuleType.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/meta/ModuleType.java
@@ -34,6 +34,11 @@ public enum ModuleType {
     /** String representation of the module type. */
     private final String label;
 
+    /**
+     * Creates a new instance.
+     *
+     * @param label label of module
+     */
     ModuleType(String label) {
         this.label = label;
     }

